web/src/features/panels/search-panel/getSearchData.ts

```diff
@@ -57,3 +118,40 @@ export const getFilteredList = (
     zoneId: zoneKey as keyof GridState,
   }));
 };
+
+// Filter solar assets based on search term
+export const getFilteredSolarAssets = (
+  searchTerm: string,
+  solarAssets: SolarAssetRowType[]
+): SolarAssetRowType[] => {
+  if (!searchTerm) {
+    return [];
+  }
+
+  const searchLower = searchTerm.toLowerCase();
+
+  return solarAssets.filter(
+    (asset) =>
+      asset.name.toLowerCase().includes(searchLower) ||
+      (asset.country && asset.country.toLowerCase().includes(searchLower)) ||
+      (asset.capacity && asset.capacity.toLowerCase().includes(searchLower)) ||
+      (asset.status && asset.status.toLowerCase().includes(searchLower))
+  );
+};
+
+// Combined search function that returns both zone and solar asset results
+export const getCombinedSearchResults = (
+  searchTerm: string,
+  zoneData: Record<string, ZoneRowType>,
+  solarAssets: SolarAssetRowType[]
+): SearchResultType[] => {
+  if (!searchTerm) {
+    return [];
+  }
+
+  const filteredZones = getFilteredList(searchTerm, zoneData);
+  const filteredSolarAssets = getFilteredSolarAssets(searchTerm, solarAssets);
+
+  // Combine results - zones first, then solar assets
+  return [...filteredZones, ...filteredSolarAssets];
+};
```

Comment:
The above function should just be updated to include these directly as we are duplicating logic and doing work twice now, as well as unpacking and re-packing two potentially large arrays. If we add more assets like wind turbines in the future this is going to be a real problem.